src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -293,48 +292,57 @@ public DateTime(int year, int month, int day, int hour, int minute, int second,
             if (second != 60 || !SystemSupportsLeapSeconds)
             {
                 ulong ticks = calendar.ToDateTime(year, month, day, hour, minute, second, millisecond).UTicks;
-                _dateData = ticks | ((ulong)kind << KindShift);
+                _dateData = ticks | ((ulong)(uint)kind << KindShift);
```

Comment:
If there's an improvement to be gained from the intermediate cast, shouldn't the JIT be able to just generate better code accordingly, given it knows from L290 that `kind` is in range of a uint?
cc: @egorbo

Reply:
@stephentoub unfortunately, it's not yet handled by JIT for shifts (just checked), but it should be trivial to implement. Minimal [repro](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGueBXADYQAdgHMGAMQgRyACgCWIjAwQBKPt07UeuhgoBmDOXP5KMahAwA8DcgAYNOvVs0uexAOzGhosWrkra1s6NQBuN10AX0iPb3sI524YpN5U33EpGVJFZVUnXW13fSMTM2VLGztHWNdU4q8fYXEA03NK4IZQxOKUhvienhSooA=). I'll take a look.

Reply:
It's not the shift but extension to `ulong` before shifting: [repro](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGueBXADYQAdgHMGAMQgRyACgCWIjAwQBKPt07UeuhgoBmDOXP5KMahAwA8DcgAYNOvVs0uexAOzGhosZZtbJHsAbjddAF9wj29QtyjnbjdfcSkZUkVlVSddbXd9IxMzZQDbBxz3PPyYn2FxNVNzUqC4xL0E/K8GVsiaCKA) should look like that.